plugins/copr.py

```diff
@@ -479,6 +480,38 @@ def _download_repo(self, project_name, repo_filename, chroot=None):
         shutil.copy2(f.name, repo_filename)
         os.chmod(repo_filename, stat.S_IRUSR | stat.S_IWUSR | stat.S_IRGRP | stat.S_IROTH)
 
+    def _runtime_deps_warning(self, copr_username, copr_projectname):
+        self.base.reset(repos=True)
+        self.base.read_all_repos()
+
+        repo = self._get_copr_repo(self._sanitize_username(copr_username), copr_projectname)
+
+        runtime_deps = []
+        for repo_id in repo.cfg.sections():
+            if repo_id.startswith("copr:"):
+                continue
+            runtime_deps.append(repo_id)
+
+        if not runtime_deps:
+            return
+
+        print(_("\nThe Copr project {0} uses the following repositories as runtime dependencies:"
+                .format(copr_projectname)))
+        dep_idx = 1
+        for repo_id in runtime_deps:
+            print("\n{0}. [{1}]".format(dep_idx, repo_id))
+            print("   baseurl={0}".format(repo.cfg.getValue(repo_id, "baseurl")))
+            dep_idx += 1
+
+        msg = _("\nDo you want to keep them enabled?")
```

Comment:
nitpick comment: should this be "Do you want to enable it?"

Reply:
It's one I requested, to make the output with separated blocks.
The newline can though be placed on a different place in code of course..

Reply:
The output looks likes this:

```
Do you really want to enable localhost:5000/dturecek/test-dependencies? [y/N]: y
Repository successfully enabled.

The Copr project test-dependencies uses the following repositories as runtime dependencies:

1. [coprdep:localhost:dturecek:foo]
   baseurl=http://backend_httpd:5002/results/dturecek/foo/fedora-$releasever-$basearch/
...
Do you want to keep them enabled? [y/N]:
```

so it seemed weird to me to first write that the repo was enabled and then ask if you want to enable the dependencies. However, I don't mind to change the message if you find it confusing.

Reply:
I am enabling "dturecek/test-dependencies" and I do not have "dturecek:foo" enabled (yet). So it does not have the sense to ask to **keep** them enabled. It does not have a sense from the language point of view.

Reply:
I get your point. However, when you are enabling `test-dependencies`, we ask if you really want to enable it. Then, it is enabled (and thus `foo` is enabled as well) and then we ask if you want to keep it enabled. So I'd argue it makes sense.

Anyway, I see that it is confusing and I don't really mind to change the message.

Reply:
I agree that we need better text.  Proposal? E.g. this?
```
Maintainer of the enabled Copr repository decided to make it dependant on
other repositories.  Such repositories are usually necessary for
successful installation of RPMs from the main Copr repository (runtime
dependencies provided).

Be aware that the note about quality and bug-reporting above applies here
too, Fedora Project doesn't control the content.  Please review the list:

... the list ...

Do you want to enable the repositories, too? [y/N]:
```

Sure, ... if something goes wrong (CTRL+C, e.g.) the repositories will
stay enabled.  But that's not a big issue.

Reply:
```
Do you really want to enable localhost:5000/dturecek/test-dependencies? [y/N]: y
Repository successfully enabled.
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ I read this that the dturecek/test-dependencies ONLY is enabled.

The Copr project test-dependencies uses the following repositories as runtime dependencies:
^^^^^^^^^^^^^^ this did not informed me that by enabling dturecek/test-dependencies the deps were enabled on MY machine too.

1. [coprdep:localhost:dturecek:foo]
   baseurl=http://backend_httpd:5002/results/dturecek/foo/fedora-$releasever-$basearch/
...
Do you want to keep them enabled? [y/N]:
```

Something like:
```
The Copr project X uses the following repositories as runtime dependencies:

1. XXXXX

These repositories have been automatically enabled.  <------- This is important

Do you want to keep them enabled? [Y/n]: <------- also note the flip in default, most people will want it enabled.
```

I will vote even for something like:
```
The Copr project X uses the following repositories as runtime dependencies:

1. XXXXX

These repositories have been automatically enabled.
```
I.e. no asking about enabling or disabling. I think most people will keep it enabled. The admin just wants to be informed of what is happening.

Reply:
> also note the flip in default

I think the default is correct (y) but we should swap the help string
`s/yN/Yn/`.  Now I understand why I was confused about the default before
....

> These repositories have been automatically enabled.
> I.e. no asking about enabling or disabling

Please no, that's what we have `dnf -y` for.  This feature brings another
level of trust.

You could as well claim that we should drop the first question because
all the people doing 'dnf copr enable ...' really want to have the project
enabled, but we still print the question because we need to have it
printed ...

Which makes me think that we should provide a warning that even the
dependencies are not maintained by Fedora Project.  This is IMO a must.

Reply:
I mean, I like this variant, thanks!
```
These repositories have been automatically enabled.

Do you want to keep them enabled? [Y/n]:
```
But we still need to tell the people why they need them, and what are
the cons.